lib/matplotlib/image.py

```diff
@@ -554,9 +553,13 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
             else:
                 if A.ndim == 2:  # _interpolation_stage == 'rgba'
                     self.norm.autoscale_None(A)
-                    A = self.to_rgba(A)
-                if A.shape[2] == 3:
-                    A = _rgb_to_rgba(A)
+                    A = self.to_rgba(A, alpha=self.get_alpha())
+                elif A.shape[2] == 3:
+                    A = _rgb_to_rgba(A, alpha=self.get_alpha())
+                elif A.shape[2] == 4:
+                    array_alpha = self.get_alpha()
+                    if array_alpha is not None and np.ndim(array_alpha) != 0:
+                        A[:, :, 3] *= array_alpha  # blend alphas of image and param
```

Comment:
This modifies `A` in place where as in the other branches we get a copy of `A` back.  Because it modifies in in place and we cache `A` higher up in the call stack it will apply the alpha everytime it draws.